pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1550,32 +1638,47 @@ func doAppNetworkModifyAllUnderlayNetworks(
 	config types.AppNetworkConfig,
 	oldConfig types.AppNetworkConfig,
 	status *types.AppNetworkStatus,
-	ipsets []string) {
+	intfList []bool, ipsets []string) error {
 
 	for i := range config.UnderlayNetworkList {
 		log.Tracef("handleModify ulNum %d\n", i)
 		ulConfig := &config.UnderlayNetworkList[i]
 		oldulConfig := &oldConfig.UnderlayNetworkList[i]
 		ulStatus := &status.UnderlayNetworkList[i]
-		doAppNetworkModifyUnderlayNetwork(
-			ctx, status, ulConfig, oldulConfig, ulStatus, ipsets, false)
+		// the underlay network associationhave changed
+		if intfList[i] {
+			// the underlay interface has already been deactivated,
+			// activate it now
+			if err := appNetworkUNetActivate(ctx, config, status,
+				ulConfig, ulStatus, ulStatus.Vif, ulStatus.Mac,
+				ipsets); err != nil {
+				return err
+			}
+		} else {
+			if err := doAppNetworkModifyUNetAcls(ctx, status, ulConfig,
+				oldulConfig, ulStatus, ipsets, false); err != nil {
+				return err
+			}
+		}
 	}
 	publishAppNetworkStatus(ctx, status)
+	return nil
 }
 
-func doAppNetworkModifyUnderlayNetwork(
+func doAppNetworkModifyUNetAcls(
```

Comment:
The name change is fine, but adding an error return makes no sense because 1) it never returns an error and 2) only one of the callers checks for an error return. So the content of this function can be restored to the original.